### DISCUSSION TITLE
IME problem

### DISCUSSION THREAD

**Discussion by OmeletWithoutEgg** [wezterm/wezterm#2929] [Q&A]
In the past few weeks I found that sometimes my IME is not working in wezterm. I'm using the ibus input method, and sometimes it just stops working (I cannot find the reason or any related action).

I've set the config as follows.

```lua
return {
    ...
    use_ime = true,
    xim_im_name = 'ibus',
    ...
}
```

Also the environment variables are set, too.

```bash
# ibus environment variable
export GTK_IM_MODULE="ibus"
export QT_IM_MODULE="ibus"
export XMODIFIERS="@im=ibus"
```

The systemd unit I use is

```service
[Unit]
Description=Intelligent Input Bus
Documentation=man:ibus-daemon(1)
Conflicts=fcitx.service

[Service]
ExecStart=/usr/bin/ibus-daemon --replace --xim
ExecReload=/usr/bin/ibus restart
ExecStop=/usr/bin/ibus exit
Environment=GTK_IM_MODULE=ibus XMODIFIERS="@im=ibus" QT_IM_MODULE=ibus

[Install]
WantedBy=default.target
# https://github.com/terlar/dotfiles/tree/master/systemd/.config/systemd/user
```

And I don't know how to get deeper or how to reproduce it stably.

P.S. The IME seems to be working well in KDE konsole.
P.S.2 There are no error or warning in ibus log.

**Comment by wez** [comment]
What version of wezterm are you running?

This issue:
* https://github.com/wez/wezterm/issues/2819

was recently fixed, which allows wezterm to recover from a hung IME (any XIM IME, not just fcitx). Previously, wezterm would stop accepting input if the IME encountered a problem or was restarted. 

Please try a nightly build to see if that improves things!

**Reply by OmeletWithoutEgg** [reply]
I was using the 20221119 version. I'm using nightly build now. It's hard to tell whether it improves since the problem occurs rarely. (in the 20221119 version it might break for 5 minutes a day and mysteriously fixed then, and the problem seems fixed now)

BTW, what's the recommended way to install a nightly build on archlinux? Instead of building from source, I installed it using `debtap` to convert the ubuntu 22.04 `.deb` file to `.pkg.tar.zst`.

**Reply by wez** [reply]
the `wezterm-git` AUR will build the current nightly from source: https://wezfurlong.org/wezterm/install/linux.html#arch-linux.
We no longer provide or recommend using binaries from other systems due to ABI mismatches and performance problems.

**Reply by OmeletWithoutEgg** [reply]
Thanks a lot! I thought it was not maintained as the pkgver is 20210916.

**Reply by cacilhas** [reply]
I’m trying to install `wezterm-git` from AUR, but it consistently fails to compile:
```
error: rustup could not choose a version of cargo to run, because one wasn't specified explicitly, and no default is configured.
help: run 'rustup default stable' to download the latest stable release of Rust and set it as your default toolchain.
==> ERROR: A failure occurred in prepare().
    Aborting...
Build failed with exit code 4
```

I tried it using `rustup default stable` and `rustup default nightly`, same results. 😥

**Comment by mistgc** [comment]
I have met this problem too. I found a resolution for enabling the input method in Wezterm in KDE Plasma.
For security, KDE Plasma only allows the program from the "Virtual Keyboard" to provide the Wayland input method service.

So, We have to enable `Fcitx5 keyboard` from the "Virtual Keyboard" in the KDE Setting.

> KDE System Setting -> Keyboard -> Vitural Keyboard -> Select Fcitx 5
